flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/health/OperatorHealthServiceTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.health;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.kubernetes.operator.config.FlinkConfigManager;
+import org.apache.flink.kubernetes.operator.config.KubernetesOperatorConfigOptions;
+
+import org.junit.jupiter.api.Test;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** Test for Health probe and endpoint logic. */
+public class OperatorHealthServiceTest {
+
+    @Test
+    public void testHealthService() throws Exception {
+        var conf = new Configuration();
+        conf.set(KubernetesOperatorConfigOptions.OPERATOR_HEALTH_PROBE_PORT, 1234);
```

Comment:
We might need an ephemeral port here, similar to how mock servers are used in unit tests.

Reply:
not if this is the only test using it right?

Reply:
but I could try to change it

Reply:
any other service could use this on a host machine

Reply:
makes sense, fixed